src/state/slices/txHistorySlice/txHistorySlice.ts

```diff
@@ -225,76 +225,62 @@ export const txHistoryApi = createApi({
         return { data: [] }
       },
     }),
-    getAllTxHistory: build.query<Transaction[], AccountId[]>({
-      queryFn: async (accountIds, { dispatch, getState }) => {
-        if (!accountIds.length) {
-          return { error: { data: 'getAllTxHistory: no account ids provided', status: 400 } }
-        }
+    getAllTxHistory: build.query<Transaction[], AccountId>({
```

Comment:
this now takes a single `AccountId` and we dispatch multiple times instead of waiting on all to upsert